### PR TITLE
Closes #182 Accounting Engine Unit test fix

### LINKED ISSUE #182: Unit Test: Accounting Engine
Depends on #214 

All tests relating to accounting engine tests should pass.

Main goal here to fix math now that surplus transfering is using WAD units.
Verify the correct amount of surplus is being transfered
`auctionSurplus` tests should be passing

### PR DESCRIPTION
**Pull request by St4rgarden**
Closes #182 

All accounting engine tests are now working as intended.

Next steps would probably be to pull anymore changes into AccountingEngine; then check coverage on Accounting Engine and complete 100% full coverage unit testing.

The absolute best way to do this is:

## Coverage Reports
[Foundry Coverage Reports To HTML](https://medium.com/@rohanzarathustra/forge-coverage-overview-744d967e112f)

Generate lcov report: 
`forge coverage --report lcov`

Convert the lcov report to HTML ( You may need to install genhtml first) : 
`genhtml -o report --branch-coverage lcov.info`

Navigate to the "report" folder, open and preview the file 'index.html' for an interactive HTML coverage report.

### CODE REVIEW

**Comment by St4rgarden** [comment]
There is also one detail I would like to flag which is the tests were expecting a particular error which is present in IAccountingEngine but it's not present in our AccountingEngine contract.

I am uncertain why this error was removed; or what it's original intended purpose was - but I recommend a quick investigation.

**Review by daopunk — APPROVED** [review]

**Comment by daopunk** [comment]
> There is also one detail I would like to flag which is the tests were expecting a particular error which is present in IAccountingEngine but it's not present in our AccountingEngine contract.
> 
> I am uncertain why this error was removed; or what it's original intended purpose was - but I recommend a quick investigation.

Can you compare with HAI repo and investigate. We can always add in our own error on the interface.

**Comment by St4rgarden** [comment]
> > IAccountingEngine but it's not present in our AccountingEngine contract.
> > I am uncertain why this error was removed; or what it's original intended purpose was - but I recommend a quick investigation.
> 
> Can you compare with HAI repo and investigate. We can always add in our own error o

Sure I can take a look tomorrow =-)

**Comment by St4rgarden** [comment]
> > There is also one detail I would like to flag which is the tests were expecting a particular error which is present in IAccountingEngine but it's not present in our AccountingEngine contract.
> > I am uncertain why this error was removed; or what it's original intended purpose was - but I recommend a quick investigation.
> 
> Can you compare with HAI repo and investigate. We can always add in our own error on the interface.

Sure you can see the error here in HAI's AccountingEngine:

```
/// @inheritdoc IAccountingEngine
  function auctionSurplus() external returns (uint256 _id) {
    if (_params.surplusIsTransferred == 1) revert AccEng_SurplusAuctionDisabled();
    if (_params.surplusAmount == 0) revert AccEng_NullAmount();
    if (block.timestamp < lastSurplusTime + _params.surplusDelay) revert AccEng_SurplusCooldown();

    uint256 _coinBalance = safeEngine.coinBalance(address(this));
    uint256 _debtBalance = safeEngine.debtBalance(address(this));
    (_coinBalance, _debtBalance) = _settleDebt(_coinBalance, _debtBalance, _unqueuedUnauctionedDebt(_debtBalance));

    if (_coinBalance < _debtBalance + _params.surplusAmount + _params.surplusBuffer) {
      revert AccEng_InsufficientSurplus();
    }

    _id = surplusAuctionHouse.startAuction({_amountToSell: _params.surplusAmount, _initialBid: 0});

    lastSurplusTime = block.timestamp;
    emit AuctionSurplus(_id, 0, _params.surplusAmount);
  }
  ```
  
  The param flag `_params.surplusIsTransferred` is used to toggle between activating two functions.  If `surplusIsTransferred` is true; then the function `transferExtraSurplus()` is active; and the function `auctionSurplus()` is deactivated ... and vice versa.

File: src/contracts/AccountingEngine.sol [hunk 1]
```diff
@@ -218,7 +218,7 @@ contract AccountingEngine is Authorizable, Modifiable, Disableable, IAccountingE
       });
 
       lastSurplusTime = block.timestamp;
-      emit AuctionSurplus(_id, 0, _params.surplusAmount.wmul(ONE_HUNDRED_WAD - _params.surplusTransferPercentage));
+      emit AuctionSurplus(_id, 0, _params.surplusAmount.wmul(WAD - _params.surplusTransferPercentage));
```
Comment (pi0neerpat) [inline src/contracts/AccountingEngine.sol:221]: good

File: test/testlocal/nft/anvil/NFT.t.sol [hunk 2]
```diff
@@ -233,8 +233,8 @@ contract NFTAnvil is AnvilFork {
    * test locking collateral
    */
 
-  function test_allowSAFE(uint256 cTypeIndex, uint8 ok) public {
-    vm.assume(ok < 2);
+  function test_allowSAFE(uint256 cTypeIndex, bool ok) public {
+    // vm.assume(ok < 2);
```
Comment (pi0neerpat) [inline test/testlocal/nft/anvil/NFT.t.sol:237]: Can you update this check for the new bool type?

**Review by pi0neerpat — CHANGES_REQUESTED** [review]

File: test/testlocal/nft/anvil/NFT.t.sol (hunk 2 above)
Comment (St4rgarden) [reply test/testlocal/nft/anvil/NFT.t.sol:237]: Sure.

Comment (St4rgarden) [reply test/testlocal/nft/anvil/NFT.t.sol:237]: I'll push up the change - although I am curious why we would make this change during the AccountingEngine test fixes?

**Review by pi0neerpat — APPROVED** [review]

**Pull request merged**